emoji/src/main/java/chat/rocket/android/emoji/CategoryPagerAdapter.kt

```diff
@@ -1,48 +1,44 @@
-package chat.rocket.android.widget.emoji
+package chat.rocket.android.emoji
 
-import androidx.viewpager.widget.PagerAdapter
-import androidx.recyclerview.widget.DefaultItemAnimator
-import androidx.recyclerview.widget.GridLayoutManager
-import androidx.recyclerview.widget.RecyclerView
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.TextView
-import chat.rocket.android.R
-import chat.rocket.android.util.extensions.setVisible
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.DefaultItemAnimator
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.viewpager.widget.PagerAdapter
+import kotlinx.android.synthetic.main.emoji_category_layout.view.*
 import java.util.*
 
-class CategoryPagerAdapter(val listener: EmojiKeyboardListener) : PagerAdapter() {
+internal class CategoryPagerAdapter(private val listener: EmojiKeyboardListener) : PagerAdapter() {
 
     override fun isViewFromObject(view: View, obj: Any): Boolean {
         return view == obj
     }
 
     override fun instantiateItem(container: ViewGroup, position: Int): Any {
         val view = LayoutInflater.from(container.context)
-                .inflate(R.layout.emoji_category_layout, container, false)
-        val layoutManager = GridLayoutManager(view.context, 8)
-        val recycler = view.findViewById(R.id.emojiRecyclerView) as RecyclerView
-        val adapter = EmojiAdapter(layoutManager.spanCount, listener)
-        val category = EmojiCategory.values().get(position)
-        val emojiNoRecentText : TextView = view.findViewById(R.id.text_no_recent_emoji)
-        val emojis = if (category != EmojiCategory.RECENTS) {
-            EmojiRepository.getEmojisByCategory(category)
-        } else {
-            EmojiRepository.getRecents()
-        }
-        val recentEmojiSize = EmojiRepository.getRecents().size
-        if (category == EmojiCategory.RECENTS && recentEmojiSize == 0){
-            emojiNoRecentText.setVisible(true)
-        }else{
-            emojiNoRecentText.setVisible(false)
+            .inflate(R.layout.emoji_category_layout, container, false)
+        with(view) {
+            val layoutManager = GridLayoutManager(context, 8)
```

Comment:
Shouldn't this `8` be dynamic?

Reply:
It probably shouldn't be hardcoded like this, but I don't know if we should calculate any possibly proper number of emojis in a row, since the layout should accommodate them evenly in each row.
I also think it's a good thing to have the same number of emojis per row on all screens. Thinking through it, I'll try it on an small-width emulator to see how it looks (and on a wider one).